server/models.py

```diff
@@ -13,23 +14,26 @@
 import playhouse.postgres_ext as pw_postgres
 
 
-HASHING_ALGORITHM = hashlib.sha3_512
+def hash_password(password: str) -> str:
+    """Hash a password."""
+    salt = os.urandom(32)
+    key = hashlib.pbkdf2_hmac('sha3-512', password.encode(), salt, 100_000)
+    return salt + key
 
-db = pw_postgres.PostgresqlExtDatabase(
-    config.DB_NAME, user=config.DB_USER, password=config.DB_PASSWORD
-)
 
+def check_password(password: str, hashed: str) -> bool:
+    """Check a password against a hash."""
+    salt = hashed[:32]
+    key = hashed[32:]
+    attempt_key = hashlib.pbkdf2_hmac(
+        'sha3-512', password.encode(), salt, 100_000
```

Comment:
There is pretty much no technical or security reason to ever use SHA3-256 - the hashing algorithm chosen for the standard was chosen because of paranoia and politics, and many other algorithms provide considerably better security (and at a much lower computational cost). For the purposes of this application, SHA256 will more than suffice.